configs/directories.go

```diff
@@ -19,23 +19,31 @@ package configs
 
 import (
 	"fmt"
-	"os"
 	"os/user"
 	"runtime"
 
 	"github.com/arduino/go-paths-helper"
-
 	"github.com/arduino/go-win32-utils"
+	"github.com/shibukawa/configdir"
 )
 
-// getDefaultConfigFilePath returns the default path for .cli-config.yml,
-// this is the directory where the arduino-cli executable resides.
+// getDefaultConfigFilePath returns the default path for arduino-cli.yaml. It searches the following directories for an existing arduino-cli.yaml file:
+// - User level configuration folder(e.g. $HOME/.config/<vendor-name>/<application-name>/setting.json in Linux)
+// - System level configuration folder(e.g. /etc/xdg/<vendor-name>/<application-name>/setting.json in Linux)
+// If it doesn't find one, it defaults to the user level configuration folder
 func getDefaultConfigFilePath() *paths.Path {
-	executablePath, err := os.Executable()
-	if err != nil {
-		executablePath = "."
+	configDirs := configdir.New("arduino", "arduino-cli")
+
+	// Search for a suitable configuration file
+	path := configDirs.QueryFolderContainsFile("arduino-cli.yaml")
+	if path != nil {
+		return paths.New(path.Path, "arduino-cli.yaml")
 	}
-	return paths.New(executablePath).Parent().Join(".cli-config.yml")
```

Comment:
Shall we warn that we changed the name? Maybe if an old config file is found arduino-cli should exit with an error asking to rename it?

Reply:
Or we rename it silently?

Reply:
I'd avoid such automatisms.

Reply:
Frankly we're still in alpha and no one is using it, so I wouldn't bother even returning an error